src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs

```diff
@@ -2792,21 +2792,24 @@ protected override IPointerTypeSymbol CommonCreatePointerTypeSymbol(ITypeSymbol
             return CreatePointerTypeSymbol(elementType.EnsureCSharpSymbolOrNull<ITypeSymbol, TypeSymbol>("elementType"));
         }
 
-        protected override INamedTypeSymbol CommonCreateTupleTypeSymbol(ImmutableArray<ITypeSymbol> elementTypes, ImmutableArray<string> elementNames)
+        protected override INamedTypeSymbol CommonCreateTupleTypeSymbol(INamedTypeSymbol underlyingType, ImmutableArray<string> elementNames)
         {
-            if (elementTypes.IsDefault)
+            if ((object)underlyingType == null)
             {
-                throw new ArgumentNullException(nameof(elementTypes));
+                throw new ArgumentNullException(nameof(underlyingType));
             }
 
-            if (elementTypes.Length <= 1)
+            var x = underlyingType.EnsureCSharpSymbolOrNull<INamedTypeSymbol, NamedTypeSymbol>(nameof(underlyingType));
```

Comment:
Sorry, that slipped through. The reason is that renaming that 'x' caused VS to crash and I got side-tracked to investigate that and then missed it. I'll fix.